### PR TITLE
Checkout: Update discount message when upgrading from domain to plan

### PR DESCRIPTION
**Pull request by gziolo**
Part of #11480. ~~Requires backend patch `D4628-code`.~~

We currently show this notice below the cart details to indicate the discount:

> You're saving $20! Your recent domain purchase is deducted from the price ($35.88), since the plan includes a free domain.

This is not actually right, since the discount is not related to the price of the domain. The actual details of how the discount is applied are a bit too complicated to go over in this notice. For consistency, probably best to just repeat the same message from the plan selection screen.

Also worth covering the renewal price here, to save any unpleasant surprises down the road.

> You're getting a $888 discount off the regular price of the plan ($888), because you already paid for the domain. 
> 
> The plan and the domain can be renewed together for $888 / year. 

After changes:

screen shot 2017-03-07 at 12 54 46

### Testing
1. ~~Apply backend patch `D4628-code`.~~
2. Purchase domain using domain only site flow.
3. Go to My Sites.
4. Click `Create Site` button.
5. Select plan and theme.
6. You should see checkout page with the updated discount copy.
screen shot 2017-03-07 at 12 54 46

### Review

* [x] Code
* [x] Product

### CODE REVIEW

File: client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx [hunk 1]
```diff
@@ -50,10 +51,35 @@ const CartPlanDiscountAd = React.createClass( {
...
+						"You're getting a %(discount)s discount off the regular price of the plan (%(originalPrice)s)" +
+						", because you already paid for the domain.",
+						{
+							args: {
+								discount: plan.formattedDiscount,
+								originalPrice: plan.formattedOriginalPrice
+							}
+						}
+					) }</p>
+					<p>{ translate(
```
Comment (aidvu) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: It looks bad if you use `<p>` here. Should probably follow the same pattern as below. :)

upsell message

**Comment by gziolo** [comment]
> Is there a possibility to reuse the message we return from the server for this?

I want to have it removed completely after this PR lands :)

**Comment by ranh** [comment]
Product 👍

File: client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx (hunk 1 above)
Comment (gziolo) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: it looks exactly the same when I omit `<p>` and use `<br>` as separator:

screen shot 2017-03-08 at 12 33 31

Comment (gziolo) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: What if I would set no margin for the last child?

Comment (aidvu) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: The screenshot you posted above looks good! The padding top/bottom is same. Check my screenshots.

Comment (gziolo) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: Updated with 1eab4e4ecfb829c29bc07b9889942bd9c02dbb7e :)

File: client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx [hunk 2]
```diff
@@ -16,18 +16,19 @@ import { getPlansBySite } from 'state/sites/plans/selectors';
 import { isPersonal, isPremium, isBusiness } from 'lib/products-values';
 import { shouldFetchSitePlans } from 'lib/plans';
 
-const CartPlanDiscountAd = React.createClass( {
-	propTypes: {
-		cart: React.PropTypes.object,
-		sitePlans: React.PropTypes.object
-	},
+class CartPlanDiscountAd extends Component {
+	propTypes = {
```
Comment (aidvu) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: ```Warning: propTypes was defined as an instance property on CartPlanDiscountAd. Use a static property to define propTypes instead.```

Should define as static prop outside when using `class`.

The rest is LGTM!

Comment (gziolo) [inline client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx]: I missed word static, Babel is going to take care of the rest ;)

**Pull request merged**